contracts/Honey.sol

```diff
@@ -37,15 +41,17 @@ contract Honey is IERC20 {
     event Approval(address indexed owner, address indexed spender, uint256 value);
     event Transfer(address indexed from, address indexed to, uint256 value);
     event AuthorizationUsed(address indexed authorizer, bytes32 indexed nonce);
-    event ChangeMinter(address indexed minter);
+    event ChangeIssuer(address indexed issuer);
 
-    modifier onlyMinter {
-        require(msg.sender == minter, "HNY:NOT_MINTER");
+    modifier onlyIssuer {
+        require(msg.sender == issuer, "HNY:NOT_ISSUER");
         _;
     }
 
-    constructor(address initialMinter) public {
-        _changeMinter(initialMinter);
+    constructor(address initialIssuer, ArbitrumGatewayRouter _gatewayRouter, address _gateway) public {
+        _changeIssuer(initialIssuer);
+        gatewayRouter = _gatewayRouter;
```

Comment:
I think we should have a function to update each of these addresses. Especially the `gateway` because gives us more flexibility to move into other L2 that support the same standard: https://ethereum-magicians.org/t/outlining-a-standard-interface-for-cross-domain-erc20-transfers/6151

Reply:
I've checked out the Optimism repo and I can't find any code adhering to that spec, only some basic bridging contracts that wouldn't require registering to like the Arbitrum ones. I've enquired with them though so will find out their intentions. Either way it would be useful to be able to update these incase we wanted to migrate to a different Arbitrum chain with a different `GatewayRouter`. Good shout.

Reply:
Right, the most likely will be another Arbitrum chain.